lib/qa/authorities/crossref/generic_authority.rb

```diff
@@ -0,0 +1,59 @@
+module Qa::Authorities
+  class Crossref::GenericAuthority < Base
+    include WebServiceBase
+    class_attribute :label, :identifier
+    attr_reader :subauthority
+
+    def initialize(subauthority)
+      @subauthority = subauthority
+    end
+
+    self.label = lambda do |item|
+      [item['name'],
+       item['alt-names'].blank? ? nil : "(#{item['alt-names'].join(', ')})",
+       item['location']].compact.join(', ')
+    end
```

Comment:
@geekscruff no, that looks great... except, we use YARD docs. Sorry! Here's a reference https://gist.github.com/chetan/1827484